dags/common/utils.py

```diff
@@ -321,37 +319,75 @@ def create_or_update_article(data):
         raise
 
 
-def parse_country_from_value(affiliation_value):
-    for key, val in INSTITUTIONS_AND_COUNTRIES_MAPPING.items():
-        if re.search(r"\b%s\b" % key, affiliation_value, flags=re.IGNORECASE):
-            return val
-    country = affiliation_value.split(",")[-1].strip()
-    for key, val in COUNTRIES_DEFAULT_MAPPING.items():
-        if re.search(r"\b%s\b" % key, country, flags=re.IGNORECASE):
-            return val
+def check_special_cases(normalized_value):
+    for pattern, country in SPECIAL_PATTERNS:
+        if pattern.search(normalized_value):
+            return country
+    return None
 
-    try:
-        country_code = cc.convert(country, to="iso2")
-        mapped_countries = []
-        if country_code != "not found":
-            mapped_countries = [
-                {
-                    "code": country_code,
-                    "name": cc.convert(country, to="name_short"),
-                }
-            ]
-
-        if len(mapped_countries) > 1 or len(mapped_countries) == 0:
-            raise FoundMoreThanOneMatchOrNone(affiliation_value)
-        return mapped_countries[0].get("name", "")
-    except (LookupError, FoundMoreThanOneMatchOrNone):
-        return find_country_match_from_mapping(affiliation_value)
-
-
-def find_country_match_from_mapping(affiliation_value):
-    for key in COUNTRIES_DEFAULT_MAPPING:
-        if re.search(r"\b%s\b" % key, affiliation_value, flags=re.IGNORECASE):
-            return COUNTRIES_DEFAULT_MAPPING[key]
+
+def unwrap_country_converter_result(converter_result):
+    if isinstance(converter_result, (list, tuple)):
+        for item in converter_result:
+            if isinstance(item, str) and item.lower() != "not found":
+                return item
+        return None
+    if not converter_result or str(converter_result).lower() == "not found":
+        return None
+    return converter_result
+
+
+def match_with_country_converter_results(normalized_value):
+    raw_iso2_result = cc.convert(names=normalized_value, to="ISO2")
+    iso2_code = unwrap_country_converter_result(raw_iso2_result)
+    if iso2_code:
+        raw_short_name = cc.convert(names=iso2_code, to="name_short")
+        country_short_name = unwrap_country_converter_result(raw_short_name)
+        if country_short_name:
+            return country_short_name
+    return None
+
+
+def fuzzy_search_country(normalized_value):
+    normalized_name_parts = normalized_value.split()
+    normalized_name_parts_length = len(normalized_name_parts)
+    max_phrase_length = min(6, normalized_name_parts_length)
+
+    for skip_from_end in range(normalized_name_parts_length):
```

Comment:
I might go for the longest phrases first `range(max_phrase_length, 0, -1):` and then the from start to phrase_length. WDYT?